### PR TITLE
date dropdown will open on top when no space bellow

### PR DESCRIPTION
**Pull request by wtfil**
* [x] - added `autoPosition` property to `Popup` which will allow popup to set position based on anchor placement and content size
* [x] - update position on `scroll` and `resize`

### CODE REVIEW

File: src/components/date-picker/date-picker.tsx [hunk 1]
```diff
@@ -101,6 +117,19 @@ export class DatePicker extends React.PureComponent<DatePickerProps, DatePickerS
         );
     }
 
+    private onRefCalendar = (calendar: Calendar) => {
```
Comment (tyv) [inline src/components/date-picker/date-picker.tsx]: this will now work if window will be scrolled or resized

**Comment by tyv** [comment]
If you do something named 'fix' you do the test accordingly. plz.

**Comment by wtfil** [comment]
@tyv Added tests

File: test/components/popup.spec.tsx [hunk 2]
```diff
@@ -60,8 +60,8 @@ describe('<Popup />', () => {
         ).withDriver(PopupTestDriver);
 
         return waitFor(() => {
-            expect(popup.root).to.be.present();
-            expect(popup.content[0]).to.be.present();
+            expect(popup.root).to.not.null;
+            expect(popup.content[0]).to.not.null;
         });
```
Comment (tyv) [inline test/components/popup.spec.tsx]: what is the difference?

File: test/components/popup.spec.tsx [hunk 3]
```diff
@@ -73,7 +73,7 @@ describe('<Popup />', () => {
         ).withDriver(PopupTestDriver);
         await sleep(100);
 
-        await waitFor(() => expect(popup.root).to.be.absent());
+        await waitFor(() => expect(popup.root).to.be.null);
     });
```
Comment (tyv) [inline test/components/popup.spec.tsx]: and here

File: test/components/popup.spec.tsx (hunk 2 above)
Comment (wtfil) [reply test/components/popup.spec.tsx]: there are no difference. I have beed playing with tests and forgot to cleanup. Now I restored original tests

**Comment by wtfil** [comment]
@tyv updated the doc

File: test/components/popup.spec.tsx [hunk 4]
```diff
@@ -96,7 +96,7 @@ describe('<Popup />', () => {
                 <span data-automation-id="SPAN">Popup Body</span>
             </Popup>).withDriver(PopupTestDriver);
 
-        await waitFor(() => {expect(popup.root).to.be.present(); });
+        await waitFor(() => {expect(popup.root).not.be.absent(); });
```
Comment (tyv) [inline test/components/popup.spec.tsx]: still why?

**Pull request merged**